types/lru-cache/index.d.ts

```diff
@@ -349,6 +334,51 @@ declare namespace LRUCache {
          */
         updateAgeOnGet?: boolean;
     }
+    type SafetyBounds<K, V> = LimitedByCount | LimitedBySize<K, V> | LimitedByTTL;
+
+    interface SharedOptions<K, V> {
+        /**
+         * Function that is called on items when they are dropped from the cache.
+         * This can be handy if you want to close file descriptors or do other
+         * cleanup tasks when items are no longer accessible. Called with `key, value`.
+         * It's called before actually removing the item from the internal cache,
+         * so if you want to immediately put it back in, you'll have to do that in
+         * a `nextTick` or `setTimeout` callback or it won't do anything.
+         */
+        dispose?: Disposer<K, V>;
+
+        /**
+         * The same as dispose, but called *after* the entry is completely removed
+         * and the cache is once again in a clean state
+         * It is safe to add an item right back into the cache at this point.
+         * However, note that it is *very* easy to inadvertently create infinite
+         * recursion this way.
+         * @since 7.3.0
+         */
+        disposeAfter?: Disposer<K, V>;
+
+        /**
+         * Set to true to suppress calling the dispose() function if the entry
+         * key is still accessible within the cache.
+         * This may be overridden by passing an options object to cache.set().
+         *
+         * @default false
+         */
+        noDisposeOnSet?: boolean;
+
+        /**
+         * Since 7.6.0
+         * `fetchMethod` Function that is used to make background asynchronous
+         *   fetches.  Called with `fetchMethod(key, staleValue)`.  May return a
+         *   Promise.
+         *
+         *     If `fetchMethod` is not provided, then `cache.fetch(key)` is equivalent
+         *     to `Promise.resolve(cache.get(key))`.
+         */
+        fetchMethod?: Fetcher<K, V> | null;
+    }
+
+    type Options<K, V> = SharedOptions<K, V> & DeprecatedOptions<K, V> & SafetyBounds<K, V>;
```

Comment:
Had to split up `Options` because there are 3 "strategies" of creating the cache - the developer _must_ provide one of `max`, `maxSize`, or `ttl` in the constructor. So I created 3 types for the config, represented in the `SafetyBounds` intersection.